x/henry/dust-apply/src/index.ts

```diff
@@ -0,0 +1,200 @@
+#!/usr/bin/env node
+
+import { DustAPI } from "@dust-tt/client";
+import * as fs from "fs";
+import * as path from "path";
+import * as os from "os";
+
+const CONFIG_FILE = path.join(os.homedir(), ".dust-apply.json");
+
+interface Config {
+  apiKey: string;
+  workspaceId: string;
+}
+
+function loadConfig(): Config {
+  if (!fs.existsSync(CONFIG_FILE)) {
+    console.error(
+      "Config file not found. Run: dust-apply config <apiKey> <workspaceId>"
+    );
+    process.exit(1);
+  }
+  return JSON.parse(fs.readFileSync(CONFIG_FILE, "utf-8"));
+}
+
+function saveConfig(config: Config) {
+  fs.writeFileSync(CONFIG_FILE, JSON.stringify(config, null, 2));
+}
+
+function extractConversationId(url: string): string {
+  const match = url.match(/dust\.tt\/w\/[^/]+\/assistant\/([a-zA-Z0-9]+)/);
+  if (!match)
+    throw new Error(
+      "Invalid URL: must be a dust.tt assistant URL (format: dust.tt/w/<workspace>/assistant/<id>)"
+    );
+  return match[1];
+}
+
+async function main() {
+  const [cmd, ...args] = process.argv.slice(2);
+
+  if (cmd === "config" && args.length === 2) {
+    saveConfig({ apiKey: args[0], workspaceId: args[1] });
+    console.log("Config saved");
+    return;
+  }
+
+  const config = loadConfig();
+  const api = new DustAPI(
+    { url: "https://dust.tt" },
+    { apiKey: config.apiKey, workspaceId: config.workspaceId },
+    console
+  );
+
+  if (cmd && cmd.includes("dust.tt")) {
+    const conversationId = extractConversationId(cmd);
+    const result = await api.getConversation({ conversationId });
+    if (result.isErr()) {
+      console.error("Error:", result.error);
+      process.exit(1);
+    }
+    const contents = result.value.content
+      .flatMap((cs) => cs[cs.length - 1])
+      // @ts-expect-error -- content is not typed
+      .map((c) => c.content)
+      .filter(Boolean)
+      .join("\n");
+    const lines = contents.split("\n");
+    type ChangeSuggestion = {
+      file: string;
+      suggestion: string;
+    };
+    let changeSuggestions: ChangeSuggestion[] = [];
+    let i = 0;
+    while (i < lines.length) {
+      const line = lines[i];
```

Comment:
Why depend on that? I would ask a first model to generate these so that it remains super versatile to the content of the convo?

Reply:
Wdym?

See Slack, the approach does not work reliably (and diffs either)

I will try "unified diffs"